cscs-checks/microbenchmarks/mpi/osu/osu_tests.py

```diff
@@ -233,70 +275,102 @@ def __init__(self):
 
 @rfm.simple_test
 class G2GBandwidthTest(P2PBaseTest):
-    def __init__(self):
-        super().__init__()
-        self.valid_systems = ['daint:gpu', 'dom:gpu', 'arolla:cn', 'tsa:cn']
-        self.num_gpus_per_node = 1
-        self.executable = './p2p_osu_bw'
-        self.executable_opts = ['-x', '100', '-i', '1000', '-d',
-                                'cuda', 'D', 'D']
-
-        self.reference = {
-            'dom:gpu': {
-                'bw': (8813.09, -0.05, None, 'MB/s')
-            },
-            'daint:gpu': {
-                'bw': (8765.65, -0.1, None, 'MB/s')
-            },
-            '*': {
-                'bw': (0, None, None, 'MB/s')
-            }
+    valid_systems = ['daint:gpu', 'dom:gpu', 'arolla:cn', 'tsa:cn']
+    num_gpus_per_node = 1
+    executable = './p2p_osu_bw'
+    executable_opts = ['-x', '100', '-i', '1000', '-d',
+                       'cuda', 'D', 'D']
+
+    reference = {
+        'dom:gpu': {
+            'bw': (8813.09, -0.05, None, 'MB/s')
+        },
+        'daint:gpu': {
+            'bw': (8765.65, -0.1, None, 'MB/s')
+        },
+        '*': {
+            'bw': (0, None, None, 'MB/s')
         }
+    }
+
+    @run_before('performance')
+    def set_performance_patterns(self):
         self.perf_patterns = {
             'bw': sn.extractsingle(r'^4194304\s+(?P<bw>\S+)',
                                    self.stdout, 'bw', float)
         }
+
+    @run_before('compile')
+    def set_cpp_flags(self):
+        self.build_system.cppflags = ['-D_ENABLE_CUDA_']
+
+    @run_before('compile')
+    def set_modules(self):
         if self.current_system.name in ['daint', 'dom']:
             self.num_gpus_per_node  = 1
-            self.modules = ['craype-accel-nvidia60']
             self.variables = {'MPICH_RDMA_ENABLED_CUDA': '1'}
+            if self.current_environ.name == 'PrgEnv-nvidia':
+                self.modules = ['cdt-cuda/21.05']
```

Comment:
I did not know before (sorry) but can you try with adding `cudatoolkit/21.3_11.2` to the list of modules and remove the ldflags to see if it works ?

Reply:
It is working with `cudatoolkit/21.3_11.2`, I will update the test 👍 The test is also passing without `cdt-cuda/21.05`, should I remove it as well?

Reply:
I would say yes, remove it.

Reply:
But having the `ldflags` seems a more maintainable solution for the future than hardcoding a specific `cudatoolkit` version, right?

Reply:
I agree (but then who will test the cudatoolkit/cray wrapper ?).
Both work, what do you think the user will prefer ?

Reply:
Not sure, but we need to stay consistent here. Now some tests do the `ldflags` by hand and others use the `cudatoolkit` wrapper. If you want to check that all those work, we should probably write a check under `prgenv`, and not test that under with a `microbenchmark`.